### PR TITLE
Replace Flow Router references with Flow Router Extra package

### PR DESCRIPTION
**Pull request by toinevk**
This PR updates current guide routing references from kadira's flow router to the supported flow-router-extra package. 

Addresses #667, touches on #640 and is complimentary to #801. 

Notably, I have removed references to using Flow Router with React. Packages designed for React such as React Router are better suited for the task currently (see https://github.com/VeliovGroup/flow-router/issues/40). These examples also referred to unmaintained packages, but happy to revert if necessary. 

TODO:

- [x] If this is a significant change, update [CHANGELOG.md](https://github.com/meteor/guide/blob/master/CHANGELOG.md)
- [x] Use `<h2 id="foo">` instead of `## Foo` for headers
- [x] Leave a blank line after each header

#

### CODE REVIEW

**Comment by toinevk** [comment]
@dr-dimitru this touches on your comment in https://github.com/meteor/guide/pull/640, let me know if you have any concerns.

**Comment by dr-dimitru** [comment]
@toinevk looks great, thank you. Hopefully, #640 can be merged after this PR.
Adding more seamless per-route dynamic imports usage examples.

React usage still under consideration, while it can be safely used with React, there are some caveats mentioned by some users, current status - waiting for more input from the community.

**Comment by dr-dimitru** [comment]
Hello folks,

Stumbled upon this old issue, why it's stuck?
Just an update about React — we have docs and confirmed support by community: https://github.com/VeliovGroup/flow-router/blob/master/docs/react.md

**Comment by StorytellerCZ** [comment]
Needs to be updated to fix conflicts, but would be great to merge already.
@benjamn

**Comment by dr-dimitru** [comment]
@StorytellerCZ @vparpoil let me know if you require help, or docs got to be updated

**Comment by StorytellerCZ** [comment]
Well, I don't have write access so @toinevk will have to update things. Worst case I can create a new PR with the fixes and changes from this one.

**Comment by toinevk** [comment]
Sorry - I lost track of this. I have updated to resolve the conflicts - should be good to merge.

File: content/accounts.md [hunk 1]
```diff
@@ -169,7 +169,8 @@ AccountsTemplates.configureRoute('resetPwd', {
 
 Note that we have specified a password reset route. Normally, we would have to configure Meteor's accounts system to send this route in password reset emails, but the `useraccounts:flow-routing` package does it for us. [Read more about configuring email flows below.](#email-flows)
 
-Now that the routes are setup on the server, they can be accessed from the browser (e.g. `example.com/reset-password`).  To create links to these routes in a template, it's best to use a helper method provided by the router.  For Flow Router, the `pathFor` helper for just this purpose.
+
+Now that the routes are setup on the server, they can be accessed from the browser (e.g. `example.com/reset-password`).  To create links to these routes in a template, it's best to use a helper method provided by the router.  For Flow Router, the [`arillo:flow-router-helpers`](https://github.com/arillo/meteor-flow-router-helpers/) package provides a `pathFor` helper for this purpose.  Once installed, the following is possible in a template:
```
Comment (dr-dimitru) [inline content/accounts.md]: Hello @toinevk,

Thank you for jumping on it. Got to mention here that `flow-router-extra` has built-in functionality and implies [`arillo:flow-router-helpers`](https://github.com/arillo/meteor-flow-router-helpers) and [`zimme:active-route`](https://github.com/meteor-activeroute/legacy) packages in its core.

**Comment by dr-dimitru** [comment]
Friday pin @toinevk @StorytellerCZ

**Comment by dr-dimitru** [comment]
I meant “friendly ping”, but spellcheck is right it’s Friday 😅

File: content/accounts.md (hunk 1 above)
Comment (toinevk) [reply content/accounts.md]: done

**Comment by toinevk** [comment]
Hopefully this can get merged!

**Comment by filipenevola** [comment]
Thanks @toinevk, this is published now.

**Pull request merged**